File: src/main/java/ru/mrgrd56/api/utils/ValueStreamingResponseBody.java
package ru.mrgrd56.api.utils;

import org.springframework.web.servlet.mvc.method.annotation.StreamingResponseBody;

import java.nio.charset.Charset;
import java.nio.charset.StandardCharsets;

public final class ValueStreamingResponseBody {
    private ValueStreamingResponseBody() { }

    public static StreamingResponseBody ofString(String value) {
        return ofString(value, StandardCharsets.UTF_8);
    }

    public static StreamingResponseBody ofString(String value, Charset charset) {
        return ofBytes(value.getBytes(charset));
    }

    public static StreamingResponseBody ofBytes(byte[] bytes) {
        return outputStream -> {
            outputStream.write(bytes);
            outputStream.close();
        };
    }
}

File: src/main/kotlin/ru/mrgrd56/api/utils/ValueStreamingResponseBody.kt
package ru.mrgrd56.api.utils

import org.springframework.web.servlet.mvc.method.annotation.StreamingResponseBody
import java.io.OutputStream
import java.nio.charset.Charset
import java.nio.charset.StandardCharsets

object ValueStreamingResponseBody {
    @JvmStatic
    @JvmOverloads
    fun ofString(value: String, charset: Charset = StandardCharsets.UTF_8): StreamingResponseBody {
        return ofBytes(value.toByteArray(charset))
    }

    @JvmStatic
    fun ofBytes(bytes: ByteArray): StreamingResponseBody {
        return StreamingResponseBody { outputStream: OutputStream ->
            outputStream.write(bytes)
            outputStream.close()
        }
    }
}
